lib/std/os/windows/user32.zig

```diff
@@ -71,105 +113,561 @@ pub const WM_MBUTTONDOWN = 0x0207;
+
+// === Windows ===
+
+pub const CS_VREDRAW = 0x0001;
+pub const CS_HREDRAW = 0x0002;
+pub const CS_DBLCLKS = 0x0008;
+pub const CS_OWNDC = 0x0020;
+pub const CS_CLASSDC = 0x0040;
+pub const CS_PARENTDC = 0x0080;
+pub const CS_NOCLOSE = 0x0200;
+pub const CS_SAVEBITS = 0x0800;
+pub const CS_BYTEALIGNCLIENT = 0x1000;
+pub const CS_BYTEALIGNWINDOW = 0x2000;
+pub const CS_GLOBALCLASS = 0x4000;
 
 pub const WNDCLASSEXA = extern struct {
     cbSize: UINT = @sizeOf(WNDCLASSEXA),
     style: UINT,
     lpfnWndProc: WNDPROC,
-    cbClsExtra: i32,
-    cbWndExtra: i32,
+    cbClsExtra: i32 = 0,
+    cbWndExtra: i32 = 0,
     hInstance: HINSTANCE,
     hIcon: ?HICON,
     hCursor: ?HCURSOR,
     hbrBackground: ?HBRUSH,
-    lpszMenuName: ?LPCSTR,
-    lpszClassName: LPCSTR,
+    lpszMenuName: ?[*:0]const u8,
+    lpszClassName: [*:0]const u8,
     hIconSm: ?HICON,
 };
 
-pub const POINT = extern struct {
-    x: c_long, y: c_long
+pub const WNDCLASSEXW = extern struct {
+    cbSize: UINT = @sizeOf(WNDCLASSEXW),
+    style: UINT,
+    lpfnWndProc: WNDPROC,
+    cbClsExtra: i32 = 0,
+    cbWndExtra: i32 = 0,
+    hInstance: HINSTANCE,
+    hIcon: ?HICON,
+    hCursor: ?HCURSOR,
+    hbrBackground: ?HBRUSH,
+    lpszMenuName: ?[*:0]const u16,
+    lpszClassName: [*:0]const u16,
+    hIconSm: ?HICON,
 };
 
-pub const MSG = extern struct {
-    hWnd: ?HWND,
-    message: UINT,
-    wParam: WPARAM,
-    lParam: LPARAM,
-    time: DWORD,
-    pt: POINT,
-    lPrivate: DWORD,
-};
+pub extern "user32" fn RegisterClassExA(*const WNDCLASSEXA) callconv(WINAPI) ATOM;
+pub fn registerClassExA(window_class: *const WNDCLASSEXA) !ATOM {
+    const atom = RegisterClassExA(window_class);
+    if (atom != 0) return atom;
+    return switch (GetLastError()) {
+        .CLASS_ALREADY_EXISTS => error.AlreadyExists,
+        .INVALID_PARAMETER => unreachable,
+        else => |err| return windows.unexpectedError(err),
+    };
+}
 
-pub extern "user32" fn CreateWindowExA(
-    dwExStyle: DWORD,
-    lpClassName: LPCSTR,
-    lpWindowName: LPCSTR,
-    dwStyle: DWORD,
-    X: i32,
-    Y: i32,
-    nWidth: i32,
-    nHeight: i32,
-    hWindParent: ?HWND,
-    hMenu: ?HMENU,
-    hInstance: HINSTANCE,
-    lpParam: ?LPVOID,
-) callconv(WINAPI) ?HWND;
+pub extern "user32" fn RegisterClassExW(*const WNDCLASSEXW) callconv(WINAPI) ATOM;
+pub var pfnRegisterClassExW: @TypeOf(RegisterClassExW) = undefined;
+pub fn registerClassExW(window_class: *const WNDCLASSEXA) !ATOM {
+    const function = selectSymbol(RegisterClassExW, pfnRegisterClassExW, .win2k);
+    const atom = function(window_class);
+    if (atom != 0) return atom;
+    return switch (GetLastError()) {
+        .CLASS_ALREADY_EXISTS => error.AlreadyExists,
+        .CALL_NOT_IMPLEMENTED => unreachable,
+        .INVALID_PARAMETER => unreachable,
+        else => |err| return windows.unexpectedError(err),
+    };
+}
+
+pub extern "user32" fn UnregisterClassA(lpClassName: LPCSTR, hInstance: HINSTANCE) callconv(.Stdcall) BOOL;
```

Comment:
You still used `LPCSTR` here, why here and not elsewhere?

Reply:
That's an error, too. I see it's `.Stdcall` as well, which is wrong... Ah, sorry about these mistakes. I for sure should look over everything more carefully before submitting.